### PR TITLE
tab completion

### PR DESCRIPTION
**Pull request by ewof**

### CODE REVIEW

**Review by LlmDl — COMMENTED** [review]
There are lots of unneeded formatting changes what will have to be undone. Before you get into that let me look at this further to see if its worth redoing or if tab completion is part of the local branch I have already.

**Review by LlmDl — CHANGES_REQUESTED** [review]
I've merged the command refactor PR which did not add any tab completion.

You'll have to rebase this PR to account for the conflict in the iConomy class, and also remove the unneeded formatting diffs.

**Review by LlmDl — CHANGES_REQUESTED** [review]
I think I've done my changes now you just have to get the tab completion showing the right commands for the given context:

- true/false should only show when /money hide NAME is used.

- /money grant NAME AMOUNT should have a silent option offered on the end of it.

- nothing should be offered when a command has reached its limit tab completion, currently it will offer player names over and over.

- /money top should prompt an amount instead of a name.


I don't think I've spotted any other funky things.

**Comment by ewof** [comment]
done

**Review by LlmDl — CHANGES_REQUESTED** [review]
Found the following issues still present.

- /money rank # continues to show a `amount` tab completion after #.
- /money top # continues to show a `amount` tab completion after #.
- /money hide name shows amount after name instead of true|false
- /money create name shows an amount after name.
- /money remove name shows an amount after name.
- /money reset name shows an amount after name.

**Comment by ewof** [comment]
fixed

File: src/main/java/com/iConomy/MoneyCommand.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+				return null;
+			if (subCmdArg.equals("top"))
+				return Arrays.asList("<amount>");
+		} else if (args.length == 3) {
+			if (AMOUNT_CMDS.contains(subCmdArg))
+				return Arrays.asList("<amount>");
+			if (subCmdArg.equals("hide"))
+				return Arrays.asList("true", "false");
+		} else if (args.length == 4 && subCmdArg.equals("grant")) {
+				return Arrays.asList("true", "false");
```
Comment (LlmDl) [inline src/main/java/com/iConomy/MoneyCommand.java]: This isn't correct, please test all of the tab completions against the help menu to confirm it is all correct, before I retest it.

**Review by LlmDl — CHANGES_REQUESTED** [review]

**Pull request merged**